model/task/task.go

```diff
@@ -177,6 +177,9 @@ type Task struct {
 	ResetWhenFinished  bool     `bson:"reset_when_finished,omitempty" json:"reset_when_finished,omitempty"`
 	DisplayTask        *Task    `bson:"-" json:"-"` // this is a local pointer from an exec to display task
 
+	// DisplayTaskId is populated for execution tasks created after this change and is explicitly empty for non execution tasks.
+	DisplayTaskId *string `bson:"display_task_id,omitempty" json:"display_task_id,omitempty"`
```

Comment:
Can you please explain more about how  for many `tasks it will be populated but empty`? Why is that the case and can we fix that?

Reply:
Is that taken care of by simply check if = ""?

Reply:
I think this is explained in the above comment and the PR comment (i.e. that a pointer to the empty string means that no display task exists, whereas a nil pointer means that we haven't checked to populate this field yet). I can add to this comment to help it be clear.